engine/gameobject/src/gameobject/gameobject.cpp

```diff
@@ -249,7 +249,7 @@ namespace dmGameObject
     Result SetCollectionDefaultCapacity(HRegister regist, uint32_t capacity)
     {
         assert(regist != 0x0);
-        if(capacity >= INVALID_INSTANCE_INDEX-1)
+        if(capacity >= INVALID_INSTANCE_INDEX - 1 || capacity <= 0)
```

Comment:
`<= 0` is invalid value as well

Reply:
It cannot be < 0 though, as it's unsigned.
I imagine the compilation warns about this?

Change to "== 0"

Reply:
Fixed